microcosm_daemon/daemon.py

```diff
@@ -90,24 +90,29 @@ def run(self):
         Run the daemon.
 
         """
-        self.parser = self.make_arg_parser()
-        self.args = self.parser.parse_args()
+        parser = self.make_arg_parser()
+        args = parser.parse_args()
 
-        if self.args.processes < 1:
-            self.parser.error("--processes must be positive")
-        elif self.args.processes == 1:
-            runner = SimpleRunner(self, self.args)
+        if args.processes < 1:
+            parser.error("--processes must be positive")
+        elif args.processes == 1:
+            runner = SimpleRunner(self)
         else:
-            runner = ProcessRunner(self.args.processes, self, self.args)
+            runner = ProcessRunner(args.processes, self)
```

Comment:
Is there an argument for always running with `ProcessRunner` to avoid multiple code paths? Is it less performant / more complicate and therefore not worth it?